tests/src/webhooks/events/invoices/InvoiceCompletedHandler.test.ts

```diff
@@ -250,4 +250,226 @@ describe('Testing the handler when an invoice is completed', () => {
+          maxSpaceBytes: lifetimeMockedMaxSpaceBytes,
+          tier: mockedLifetimeTier,
+        });
+        const applyDriveFeaturesSpy = jest.spyOn(tiersService, 'applyDriveFeatures').mockResolvedValue();
+        const applyVpnFeaturesSpy = jest.spyOn(tiersService, 'applyVpnFeatures').mockResolvedValue();
+
+        const handleNewProduct = invoiceCompletedHandler['handleNewProduct'].bind(invoiceCompletedHandler);
+        await handleNewProduct({
+          user: {
+            ...mockedUser,
+            email: mockedCustomer.email as string,
+          },
+          customer: mockedCustomer,
+          isLifetimePlan: mockedIsLifetimePlan,
+          productId: mockedProductId,
+          totalQuantity,
+          tier: mockedTier,
+        });
+
+        expect(determineLifetimeConditionsSpy).toHaveBeenCalledWith(mockedUser, mockedProductId);
+        expect(applyDriveFeaturesSpy).toHaveBeenCalledWith(
+          {
+            ...mockedUser,
+            email: mockedCustomer.email as string,
+          },
+          mockedCustomer,
+          totalQuantity,
+          mockedLifetimeTier,
+          expect.anything(),
+          lifetimeMockedMaxSpaceBytes,
+        );
+        expect(applyVpnFeaturesSpy).toHaveBeenCalledWith(
+          {
+            ...mockedUser,
+            email: mockedCustomer.email as string,
+          },
+          mockedLifetimeTier,
+        );
+      });
+
+      test('When an error occurs while determining the lifetime conditions, then a log is printed but the flow continues with the default tier', async () => {
+        const unexpectedError = new Error('Unexpected error');
+        const mockedUser = getUser();
+        const mockedCustomer = getCustomer({
+          email: 'test@inxt.com',
+        });
+        const mockedIsLifetimePlan = true;
+        const mockedProductId = getProduct({}).id;
+        const totalQuantity = 1;
+        const mockedTier = newTier();
+        const determineLifetimeConditionsSpy = jest
+          .spyOn(determineLifetimeConditions, 'determine')
+          .mockRejectedValue(unexpectedError);
+        const applyDriveFeaturesSpy = jest.spyOn(tiersService, 'applyDriveFeatures').mockResolvedValue();
+        const applyVpnFeaturesSpy = jest.spyOn(tiersService, 'applyVpnFeatures').mockResolvedValue();
+
+        const handleNewProduct = invoiceCompletedHandler['handleNewProduct'].bind(invoiceCompletedHandler);
+        await handleNewProduct({
+          user: {
+            ...mockedUser,
+            email: mockedCustomer.email as string,
+          },
+          customer: mockedCustomer,
+          isLifetimePlan: mockedIsLifetimePlan,
+          productId: mockedProductId,
+          totalQuantity,
+          tier: mockedTier,
+        });
+
+        expect(determineLifetimeConditionsSpy).toHaveBeenCalledWith(mockedUser, mockedProductId);
+        expect(applyDriveFeaturesSpy).toHaveBeenCalled();
```

Comment:
Try to match the call parameters, otherwise you do not know if that 'default tier' you have mentioned is the one used to 'continue the flow'